test/lit.cfg

```diff
@@ -540,6 +540,11 @@ config.swift_driver_test_options += os.environ.get('SWIFT_DRIVER_TEST_OPTIONS',
 config.swift_ide_test_test_options += os.environ.get('SWIFT_IDE_TEST_TEST_OPTIONS', '')
 config.sil_test_options = os.environ.get('SIL_TEST_OPTIONS', '')
 
+# Check if we overrode the linker; if we do, set a feature to say that, which
+# lets us disable tests that rely on the driver choosing the linker.
+if re.search(r'(?:^|[ \t])-use-ld=[^ \t]*', config.swift_driver_test_options):
+    config.available_features.add('linker_overridden')
```

Comment:
This seems overly defensive, particularly since the tests handle multiple linkers. Any particular problem you're trying to avoid with linker overrides?

Reply:
Not all of the tests handle overriding the linker, which is the point of this. Specifically, the tests I tagged with `linker_overridden` won't work with `-use-ld=<whatever>` in `SWIFT_DRIVER_TEST_OPTIONS` because they rely on the fact that the driver will choose an appropriate linker based on either the specified target or the LTO setting. If you force it with `-use-ld=<something>`, then it will try to use that thing instead of letting the driver pick, and those tests will then fail.

Reply:
Oh, I see: these are tests that already pass `-use-ld` for various reasons, so that expected linker cannot be overridden, got it.